lib/_http_outgoing.js

```diff
@@ -509,6 +524,14 @@ OutgoingMessage.prototype.end = function(data, encoding, callback) {
     this.once('finish', callback);
 
   if (!this._header) {
+    if (data) {
+      if (typeof data === 'string')
+        this._contentLength = Buffer.byteLength(data, encoding);
+      else
+        this._contentLength = data.length;
+    } else {
+      this._contentLength = 0;
+    }
```

Comment:
Can you create the _contentLength field in the OutgoingMessage constructor?  Adding properties post facto causes a hidden class transition.